### PR TITLE
New option: bucket_size

### PR DESCRIPTION
**Pull request by OttaviaB**
### Relevant Issue

#2111
### Details

This pull request adds the new `bucket_size` option, which allows to set the size of the mounted bucket advertised by s3fs to a user-defined value.

While the previous behaviour is preserved as the default, this new option can be used when accurate metrics are needed.

### CODE REVIEW

File: src/s3fs.cpp [hunk 1]
```diff
@@ -2808,15 +2810,15 @@ static int s3fs_statfs(const char* _path, struct statvfs* stbuf)
...
+    stbuf->f_blocks = bucket_size;
+    stbuf->f_bfree = stbuf->f_blocks;
 #elif defined(__APPLE__)
-    stbuf->f_blocks = UINT32_MAX;
-    stbuf->f_bfree  = UINT32_MAX;
-    stbuf->f_files  = UINT32_MAX;
-    stbuf->f_ffree  = UINT32_MAX;
+    stbuf->f_blocks = bucket_size;
+    stbuf->f_bfree  = stbuf->f_blocks;
+    stbuf->f_bfree  = stbuf->f_blocks;
```
Comment (ggtakec) [inline src/s3fs.cpp:2818]: Isn't this a mistake in `stbuf->f_files`?

**Review by ggtakec — CHANGES_REQUESTED** [review]
Please check comments

**Comment by ggtakec** [comment]
@OttaviaB Thank you for your kindness!
I agree with this amendment, but it seems that one correction is required.

File: src/s3fs.cpp (hunk 1 above)
Comment (OttaviaB) [reply src/s3fs.cpp]: Yes, definitely. I have just pushed a new commit where f_files and f_ffree are left defined as UINT32MAX as before.

**Comment by ggtakec** [comment]
@OttaviaB Thanks for your contribution.
I'll merge this PR soon.

**Pull request merged**